.github/workflows/pr.yml

```diff
@@ -0,0 +1,25 @@
+name: PR testing
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '8', '11' ]
+    name: Java ${{ matrix.java }} tests
+    steps:
+      - uses: actions/checkout@master
+      - name: setup java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java }}
+      - name: setup lein
+        run: |
+          mkdir -p local/bin
+          curl --output local/bin/lein https://raw.githubusercontent.com/technomancy/leiningen/2.9.1/bin/lein
+          chmod +x local/bin/lein
+          sudo local/bin/lein
```

Comment:
Sort of a trivial point here but I'll ask anyway: is it possible to not use 'sudo'?